lib/PuppeteerSharp.Tests/FrameTests/GoToTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PuppeteerSharp.Tests.FrameTests
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class GoToTests : PuppeteerPageBaseTest
+    {
+        public GoToTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldNavigateSubFrames()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
+            Assert.Contains("/frames/one-frame.html", Page.Frames[0].Url);
+            Assert.Contains("/frames/frame.html", Page.Frames[1].Url);
+            var response = await Page.Frames[1].GoToAsync(TestConstants.EmptyPage);
+            Assert.Equal(HttpStatusCode.OK, response.Status);
+        }
+
+        [Fact]
+        public async Task ShouldRejectWhenFrameDetaches()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
+            Server.SetRoute("/empty.html", context => Task.Delay(10000));
+            var navigationTask = Page.Frames[1].GoToAsync(TestConstants.EmptyPage);
```

Comment:
@kblok I seem to be stuck again. For some reason, the test hangs at `Server.WaitForRequest()`. I figured it should just wait 10 seconds, but it never gets past that. I'm thinking I messed up the frame navigation somewhere, but the first test seems to work fine. Any suggestions? Thanks

Reply:
I'll take a look at it

Reply:
The problem is that the request is getting to the server before we start waiting for the request. This will work
```cs 
[Fact]
public async Task ShouldRejectWhenFrameDetaches()
{
    await Page.GoToAsync(TestConstants.ServerUrl + "/frames/one-frame.html");
    Server.SetRoute("/empty.html", context => Task.Delay(10000));
    var waitForRequestTask = Server.WaitForRequest("/empty.html");
    var navigationTask = Page.Frames[1].GoToAsync(TestConstants.EmptyPage);
    await waitForRequestTask;
    await Page.QuerySelectorAsync("iframe").EvaluateFunctionAsync("frame => frame.remove()");
    var exception = await Assert.ThrowsAsync<NavigationException>(async () => await navigationTask);
    Assert.Equal("Navigating frame was detached", exception.Message);
}
```